builtin/commit.c

```diff
@@ -101,6 +101,7 @@ static int all, also, interactive, patch_interactive, only, amend, signoff;
 static int edit_flag = -1; /* unspecified */
 static int quiet, verbose, no_verify, allow_empty, dry_run, renew_authorship;
 static int config_commit_verbose = -1; /* unspecified */
+static int config_commit_allow_empty = 0; /* unspecified, 0 for bool*/
```

Comment:
A `static` is initalized as all zeroes by default, and it is our convention to skip the explicit initialization (see e.g. the `quiet` variable two lines above).

However, this is not really what we want here. We want to initialize it to `-1`, just like `config_commit_verbose` in the line above, to indicate that it was unspecified.